packages/Search/src/details/DTK_DetailsTreeConstruction_decl.hpp

```diff
@@ -36,22 +40,24 @@ struct TreeConstruction
   public:
     using ExecutionSpace = typename DeviceType::execution_space;
 
-    static void calculateBoundingBoxOfTheScene(
-        Kokkos::View<Box const *, DeviceType> bounding_boxes,
-        Box &scene_bounding_box );
+    template <typename ConstViewType>
```

Comment:
I would not call it `ConstViewType` as you cannot guarantee that. Just call it `ViewType`.

Reply:
Also, would it be easier to template `TreeConstruction` on the `ViewType` instead of each individual function? It should not make a difference in compilation times.

Reply:
Will rename it to `Primitives`